UriInfo can not be injected in presence of quarkus-rest-client dependency

### Describe the bug

I has an application, which uses a filter (derived from `ContainerRequestFilter`) and adds UriInfo into it via `@Inject` annotation. When I add `quarkus-rest-client` into that application, it fails with `UnsatisfiedResolutionException` during build.

This behavior can only be observed while using the latest Quarkus, classic resteasy and rest-client extensions and `@Inject` annotation. It can not be observed with released version, reactive extensions and when UriInfo injected via `@Context` annotation.

### Expected behavior

Application should build and start properly.

### Actual behavior

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (default) on project theapp: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: jakarta.enterprise.inject.spi.DeploymentException: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type jakarta.ws.rs.core.UriInfo and qualifiers [@Default]
[ERROR] 	- java member: io.quarkus.vertx.LoggingFilter#info
[ERROR] 	- declared on CLASS bean [types=[java.lang.Object, io.quarkus.vertx.LoggingFilter], qualifiers=[@Default, @Any], target=io.quarkus.vertx.LoggingFilter]
[ERROR] 	at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1447)
[ERROR] 	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:311)
[ERROR] 	at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:158)
[ERROR] 	at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:471)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:858)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:833)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] Caused by: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type jakarta.ws.rs.core.UriInfo and qualifiers [@Default]
[ERROR] 	- java member: io.quarkus.vertx.LoggingFilter#info
[ERROR] 	- declared on CLASS bean [types=[java.lang.Object, io.quarkus.vertx.LoggingFilter], qualifiers=[@Default, @Any], target=io.quarkus.vertx.LoggingFilter]
[ERROR] 	at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:478)
[ERROR] 	at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:624)
[ERROR] 	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:299)
[ERROR] 	... 13 more

```

### How to Reproduce?

1. `git clone https://github.com/fedinskiy/reproducer -b uri-info && cd reproducer`
2. `mvn clean verify -Pclassic,client #fails`
3. `mvn clean verify -Pclient,classic -Dquarkus.platform.version=3.3.2 -Dquarkus.platform.group-id=io.quarkus.platform #works`
4.  `mvn clean verify -Pclassic #works`
5.  `mvn clean verify -Preactive #works`
6. Change this line[1] from `@Inject` to `@Context`, `mvn clean verify -Pclassic,client #works`
[1] https://github.com/fedinskiy/reproducer/blob/uri-info/src/main/java/io/quarkus/vertx/LoggingFilter.java#L18

### Output of `uname -a` or `ver`

6.4.14-200.fc38.x86_64

### Output of `java -version`

 Java version: 17.0.8, vendor: Eclipse Adoptium

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

718a8d5eadcdaf7c342510d6e5a4463e01fbee83

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

 Apache Maven 3.8.7 (b89d5959fcde851dcb1c8946a785a163f14e1e29)

### Additional information

_No response_

--- a/extensions/resteasy-classic/rest-client/runtime/pom.xml
+++ b/extensions/resteasy-classic/rest-client/runtime/pom.xml
@@ -25,12 +25,6 @@
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-resteasy-common</artifactId>
-            <exclusions>
-                <exclusion>
-                    <groupId>org.jboss.resteasy</groupId>
-                    <artifactId>resteasy-cdi</artifactId>
-                </exclusion>
-            </exclusions>
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>

--- a/extensions/resteasy-classic/resteasy-common/runtime/pom.xml
+++ b/extensions/resteasy-classic/resteasy-common/runtime/pom.xml
@@ -83,10 +83,6 @@
         </dependency>
         <dependency>
             <groupId>org.jboss.resteasy</groupId>
-            <artifactId>resteasy-cdi</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>org.jboss.resteasy</groupId>
             <artifactId>resteasy-json-binding-provider</artifactId>
             <optional>true</optional>
             <exclusions>

--- a/extensions/resteasy-classic/resteasy-server-common/runtime/pom.xml
+++ b/extensions/resteasy-classic/resteasy-server-common/runtime/pom.xml
@@ -26,6 +26,10 @@
             <artifactId>quarkus-resteasy-common</artifactId>
         </dependency>
         <dependency>
+            <groupId>org.jboss.resteasy</groupId>
+            <artifactId>resteasy-cdi</artifactId>
+        </dependency>
+        <dependency>
             <groupId>jakarta.validation</groupId>
             <artifactId>jakarta.validation-api</artifactId>
         </dependency>

